subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/MerkleDirectorySnapshotBuilder.java

```diff
@@ -93,7 +94,7 @@ public boolean postVisitDirectory(boolean includeEmpty) {
             hasher.putString(child.getName());
             hasher.putHash(child.getHash());
         }
-        CompleteDirectorySnapshot directorySnapshot = new CompleteDirectorySnapshot(absolutePath, name, children, hasher.hash());
+        CompleteDirectorySnapshot directorySnapshot = new CompleteDirectorySnapshot(absolutePath, name, children, hasher.hash(), AccessType.DIRECT);
```

Comment:
Can you explain in a comment why this is a `DIRECT` snapshot always?

Reply:
The latest code doesn't use `DIRECT` in this class at all.